solidity/contracts/PetalLock.sol

```diff
@@ -90,6 +90,16 @@ struct BatchExtend {
 }
 
 contract PetalLock {
+	event MadeImmutable(string fullName, uint256 tokenId, bytes contenthash, address resolutionAddress);
+
+	function formFullPathEnsNameString(string[] memory inputArray) private pure returns (string memory) {
+		bytes memory result;
+		for (uint i = inputArray.length; i > 0; i--) {
+			result = abi.encodePacked(result, inputArray[i - 1], '.');
+		}
```

Comment:
Out of curiosity, why not this, which I believe is a bit more idiomatic (doesn't result in a second glance)?
```suggestion
		for (uint i = inputArray.length - 1; i >= 0; i--) {
			result = abi.encodePacked(result, inputArray[i], '.');
		}
```